### ISSUE TITLE
Resource leak reported by Netty

### ISSUE THREAD

**Issue by esocode** [micronaut-projects/micronaut-core#3855]
### Task List

- [ ] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

Yesterday I had deployed 2 rather simple services into a Kubernetes cluster. Both provide a REST API to databases and use micronaut-data with JPA to connect to a MariaDB. The services are part of a larger deployment and had therefore not yet been used. When I looked at the container logs this morning I noticed a resource leak exception in both services that occurred several hours after deployment. Although there is no easy way to reproduce this effect I thought it might help to provide the stack trace as this could indicate a problem in long-running services.

```java
17:13:34.221 [main] INFO  io.micronaut.runtime.Micronaut - Startup completed in 4212ms. Server Running: http://api-service-869d89c6d5-7xt5v:8080
22:18:12.311 [parallel-1] ERROR i.m.s.i.n.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See https://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records:
Created at:
        io.micrometer.shaded.io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:363)
        io.micrometer.shaded.io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:187)
        io.micrometer.shaded.io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:173)
        io.micrometer.shaded.io.netty.buffer.AbstractByteBufAllocator.buffer(AbstractByteBufAllocator.java:107)
        io.micrometer.shaded.reactor.netty.NettyOutbound.lambda$sendString$5(NettyOutbound.java:300)
        io.micrometer.shaded.reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:100)
        io.micrometer.shaded.reactor.core.publisher.FluxOnBackpressureLatest$LatestSubscriber.drain(FluxOnBackpressureLatest.java:176)
        io.micrometer.shaded.reactor.core.publisher.FluxOnBackpressureLatest$LatestSubscriber.onNext(FluxOnBackpressureLatest.java:127)
        io.micrometer.shaded.reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:114)
        io.micrometer.shaded.reactor.core.publisher.FluxBufferPredicate$BufferPredicateSubscriber.onNextNewBuffer(FluxBufferPredicate.java:269)
        io.micrometer.shaded.reactor.core.publisher.FluxBufferPredicate$BufferPredicateSubscriber.tryOnNext(FluxBufferPredicate.java:216)
        io.micrometer.shaded.reactor.core.publisher.FluxBufferPredicate$BufferPredicateSubscriber.onNext(FluxBufferPredicate.java:186)
        io.micrometer.shaded.reactor.core.publisher.FluxFlatMap$FlatMapMain.tryEmit(FluxFlatMap.java:535)
        io.micrometer.shaded.reactor.core.publisher.FluxFlatMap$FlatMapInner.onNext(FluxFlatMap.java:999)
        io.micrometer.shaded.reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:192)
        io.micrometer.shaded.reactor.core.publisher.DirectProcessor$DirectInner.onNext(DirectProcessor.java:333)
        io.micrometer.shaded.reactor.core.publisher.DirectProcessor.onNext(DirectProcessor.java:142)
        io.micrometer.shaded.reactor.core.publisher.FluxCreate$IgnoreSink.next(FluxCreate.java:618)
        io.micrometer.shaded.reactor.core.publisher.FluxCreate$SerializedSink.next(FluxCreate.java:153)
        io.micrometer.statsd.internal.LogbackMetricsSuppressingFluxSink.lambda$next$1(LogbackMetricsSuppressingFluxSink.java:49)
        io.micrometer.core.instrument.binder.logging.LogbackMetrics.ignoreMetrics(LogbackMetrics.java:113)
        io.micrometer.statsd.internal.LogbackMetricsSuppressingFluxSink.next(LogbackMetricsSuppressingFluxSink.java:49)
        io.micrometer.statsd.internal.LogbackMetricsSuppressingFluxSink.next(LogbackMetricsSuppressingFluxSink.java:25)
        io.micrometer.statsd.StatsdGauge.poll(StatsdGauge.java:56)
        io.micrometer.statsd.StatsdMeterRegistry.poll(StatsdMeterRegistry.java:175)
        io.micrometer.statsd.StatsdMeterRegistry.lambda$startPolling$13(StatsdMeterRegistry.java:282)
        io.micrometer.shaded.reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onNext(FluxDoOnEach.java:140)
        io.micrometer.shaded.reactor.core.publisher.FluxInterval$IntervalRunnable.run(FluxInterval.java:123)
        io.micrometer.shaded.reactor.core.scheduler.PeriodicWorkerTask.call(PeriodicWorkerTask.java:59)
        io.micrometer.shaded.reactor.core.scheduler.PeriodicWorkerTask.run(PeriodicWorkerTask.java:73)
        java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
        java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
        java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
        java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        java.lang.Thread.run(Thread.java:748)
```

### Expected Behaviour

No resource leak.

### Actual Behaviour

Resource leak.

### Environment Information

- **Operating System**: Linux, Kubernetes Cluster 
- **Micronaut Version:** 2.0.1
- **JDK Version:** 8

**Comment by graemerocher** [comment]
Seems to be a Micrometer bug as it is the shaded version of Netty unrelated to Micronaut

**Comment by esocode** [comment]
I have reported this in the micrometer project:
https://github.com/micrometer-metrics/micrometer/issues/2222

**Comment by esocode** [comment]
Should be fixed in micrometer 1.5.3.
